x/protorev/keeper/posthandler.go

```diff
@@ -62,8 +62,8 @@ func (protoRevDec ProtoRevDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simu
 	}
 
 	// Delete swaps to backrun for next transaction without consuming gas
-	// from the current transaction's gas meter, but instead from a new gas meter
-	protoRevDec.ProtoRevKeeper.DeleteSwapsToBackrun(ctx.WithGasMeter(upperGasLimitMeter))
+	// from the current transaction's gas meter, but instead from an infinite gas meter.
+	protoRevDec.ProtoRevKeeper.DeleteSwapsToBackrun(ctx.WithGasMeter(sdk.NewInfiniteGasMeter()))
```

Comment:
we should definitely bound this, maybe increase by x2 or x4 or anything else 

i did some gas estimate check using our SwapLogic: https://github.com/osmosis-labs/osmosis/pull/5850/files#diff-1a552fcc0a3d36556402345e840b658a7daccb101d7f73bca9ee2e3057ade2e6R3451

and here's the output. While Protorev route building, more than 2000 ticks was crossed
```
tickSpacing = 1

swapOutGivenIn direction
TickCrossed = 1754 (from -20 to 1755) 
positions = 200 tick wide positions, ranging from [-5000 to 5000] 
NewGasCost = 22239116, OldGasCost = 22255556, DIFF = 16440

swapInGivenOut direction
TickCrossed = 998 (from 1755 to 736) 
positions = 200 tick wide positions, ranging from [-5000 to 5000] tick
NewGasCost = 22358879, OldGasCost = 22383527, DIFF = 24648
```

Reply:
Using an infinite gas meter here does not mean we are having an unbounded gas meter for the rebalancing logic itself, that (at this point) is still 50 mil gas. This change is to use an infinite gas meter at the point of deleting swaps to backrun from the kvstore such that there will never be the ability for an out of gas error during deletion.

We should not use the same gasmeter for both rebalancing and deleting, we could use an arbitrarily high gasmeter here instead if that's desired.

Reply:
Put in another way, this change has no impact on the amount of ticks that can be crossed or the desired bounding of protorev computation itself for rebalancing.

Reply:
This commit changes it to use a new 50mil gas meter: https://github.com/osmosis-labs/osmosis/pull/5841/commits/5789614f43014856ba52cd3ec282588f2c07765b

So now this can still fail a user's tx if for some reason deleting swaps from storage takes over 50mil gas (should never happen, but using any number that's not infinite gas meter runs into this issue)

Reply:
i see its the pointer reference error where "upperGasLimitMeter" turns into 0 before `DeleteSwapsToBackrun` gets called? i think it will be very helpful if we add this comment too https://github.com/osmosis-labs/osmosis/pull/5841#discussion_r1265816959

Reply:
@stackman27 Correct, this PR is purely to fix the shared reference usage. With that understanding, would you still prefer this to be a bounded gas limit? If so, I will change it to something smaller (10mil gas) just so that 50mil isn't used twice

Reply:
yea i dont think deleting is too expensive, we can just set it to 50M and add inline comment on why we did that

Reply:
comment added: https://github.com/osmosis-labs/osmosis/pull/5841/commits/b06bdb43604e3eba42ebdb71c3ee4daba78c6112